extension/intents/slideshow/contentScript.js

```diff
@@ -0,0 +1,186 @@
+/* globals communicate */
+
+this.slideshowScript = (function() {
+  let slideElements = [];
+  let currentSlideIndex = 0;
+
+  const videoSources = ["youtube.com", "vimeo.com"];
+
+  slideElements = detectAllMedia();
+  buildSlideStructure();
+
+  const lbContainer = document.body.querySelector(".fv-lightbox-container");
+  const slideContainer = document.body.querySelector(".fv-slide-image");
+
+  window.onresize = function() {
+    if (slideContainer.hasChildNodes()) {
+      if (slideContainer.firstChild.tagName === "IFRAME") {
+        slideContainer.firstChild.setAttribute(
+          "width",
+          document.documentElement.clientWidth
+        );
+        slideContainer.firstChild.setAttribute(
+          "height",
+          document.documentElement.clientHeight
+        );
+      }
+    }
+  };
+
+  function buildSlideStructure() {
+    const lightboxElement = document.createElement("div");
+    lightboxElement.className = "fv-lightbox-container";
+    lightboxElement.onclick = hideSlideShow;
+
+    const slideshowContainer = document.createElement("div");
+    slideshowContainer.className = "fv-slideshow-container";
+
+    const navbar = document.createElement("div");
+    navbar.className = "fv-navbar";
+
+    const tagClose = document.createElement("a");
+    tagClose.className = "fv-close";
+    tagClose.textContent = String.fromCharCode(0x274c);
+    tagClose.onclick = hideSlideShow;
+
+    const tagPrev = document.createElement("a");
+    tagPrev.className = "fv-prev";
+    tagPrev.textContent = String.fromCharCode(10094);
+    tagPrev.onclick = previousSlide;
```

Comment:
It doesn't _really_ matter, but it's considered better form to use `tagPrev.addEventListener("click", previousSlide);`

Reply:
@ianb I have modified all event handlers to use this form